core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/controller/promisehandlers/QueryWorkerStatisticsHandler.scala

```diff
@@ -37,4 +55,37 @@ trait QueryWorkerStatisticsHandler {
       })
     }
   }
+
+  registerHandler((msg: ControllerInitiateQueryResults, sender) => {
+    val sinkWorkers = workflow.getSinkLayers.flatMap(l => l.workers.keys).toList
+    val workers = msg.workers.getOrElse(sinkWorkers)
+
+    // send all sink worker QueryResult message
+    val requests = workers.map(worker => {
+      send(QueryWorkerResult(), worker).map(res => (worker, res))
+    })
+
+    // wait for all workers to reply, accumulate response from all workers
+    val allResponses = Future.collect(requests)
+
+    allResponses.map(responses => {
+      // combine results of all workers to a single result list of this operator
+      val operatorResultUpdate = new mutable.HashMap[String, OperatorResult]()
+      responses
+        .groupBy(workerResult => workflow.getOperator(workerResult._1).id)
+        .foreach(operatorResult => {
+          val workerResultList = operatorResult._2.flatMap(r => r._2)
+          if (workerResultList.nonEmpty) {
+            val operatorID = operatorResult._1.operator
+            val outputMode = workerResultList.head.outputMode
+            operatorResultUpdate(operatorID) =
+              OperatorResult(outputMode, workerResultList.flatMap(r => r.result).toList)
```

Comment:
Output mode is `snapshot` or `delta`. This could be hybrid (some sink outputs snapshot, some sink outputs delta). 

But we don't allow disabling progressive update for some operators yet. All sinks are going to produce the result update every second.